src/components/Form.js

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import './Form.css';
+
+export class Form extends React.Component {
+  state = {
+    task: '',
+    userName: '',
+    user: '',
+    userError: false,
+    taskError: false,
+  };
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+    const { task, userName, user } = this.state;
+    const { addUser } = this.props;
+
+    if (!task) {
+      this.setState({
+        taskError: true,
+      });
+
+      return;
+    }
+
+    if (!userName) {
+      this.setState({
+        userError: true,
+      });
+
+      return;
+    }
+
+    addUser(user, task);
+    this.setState({
+      task: '',
+      userName: '',
+      userError: false,
+      taskError: false,
+    });
+  }
+
+  handleSelect = (event) => {
+    const { users } = this.props;
+
+    this.setState({
+      [event.target.name]: event.target.value,
+    });
+    this.setState(prevState => ({
+      user: users.find(userToFind => (
+        userToFind.name === prevState.userName
+      )),
+    }));
+  }
```

Comment:
let's create one function handleChange which we can use for both input and select